### ISSUE TITLE
Named Data Sources?

### ISSUE THREAD

**Issue by metaperture** [vega/react-vega#23]
Do you have any plans to support named data sources in Vega ([link](https://vega.github.io/vega-lite/docs/data.html#named))? For large layered charts it can be much more performant.

**Comment by eharkins** [comment]
@metaperture I've been able to pass a prop (using https://github.com/vega/react-vega#approach2-use-vega-generic-class-and-pass-in-spec-for-dynamic-component) called `data` as an object with keys as the names of my data sources and values as the collections of data for each (https://github.com/vega/react-vega#data).  Is that what you were asking about?

**Comment by vaughnkoch** [comment]
@eharkins It doesn't seem to be possible to pass in multiple named datasets with the data parameter like this:

```
  const data = {
    "timeseriesA": timeseriesA,
    "timeseriesB": timeseriesB,
  }

  return (
    <VegaLite
      renderer="canvas"
      spec={spec}
      data={data}
      width={width}
      height={height}
      onNewView={vegaOnNewView}
      onParseError={onParseError}
    />
```

From code inspection and debugging, I think the 'data' parameter passed to <VegaLite> has to conform to the InlineData format: https://vega.github.io/vega-lite/docs/data.html#inline, meaning it needs to have `values` at least.

It would be great to have explicit support for vega-lite datasets. That would let you actually pass in multiple values.

**Comment by eharkins** [comment]
@vaughnkoch [here is an example of how we are doing this with react-vega and Vega (not Vega-Lite)](https://github.com/matsengrp/olmsted/blob/master/src/components/explorer/tree.js#L108-L165). According to the docs it seems the difference is the `values` key is not required in Vega while it is required in Vega-Lite.

**Comment by domoritz** [comment]
Note that this library doesn't have good support for data yet. It's still in it's early stages. If you want more control, build your own react component with Vega Embed. We would also love your help with adding features to this library since my cycles are limited.

**Comment by kristw** [comment]
This should be resolved in `react-vega@7.0.0`. I will close the ticket now but please re-open if I misunderstood or you still have issue with it.
